src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -138,23 +134,27 @@ public SortedSet(IEnumerable<T> collection, IComparer<T> comparer)
                     }
                 }
                 _count = baseSortedSet._count;
-                _version = 0;
             }
             else
-            {     //As it stands, you're doing an NlogN sort of the collection
-                List<T> els = new List<T>(collection);
-                els.Sort(_comparer);
-                for (int i = 1; i < els.Count; i++)
+            {
+                int count;
+                T[] els = EnumerableHelpers.ToArray(collection, out count);
+                if (count > 0)
                 {
-                    if (comparer.Compare(els[i], els[i - 1]) == 0)
+                    Array.Sort(els, 0, count, _comparer);
+                    int index = 1;
+                    for (int i = 1; i < count; i++)
                     {
-                        els.RemoveAt(i);
-                        i--;
+                        if (comparer.Compare(els[i], els[i - 1]) != 0)
+                        {
+                            els[index++] = els[i];
+                        }
                     }
+                    count = index;
+
+                    _root = ConstructRootFromSortedArray(els, 0, count - 1, null);
+                    _count = count;
                 }
-                _root = ConstructRootFromSortedArray(els.ToArray(), 0, els.Count - 1, null);
-                _count = els.Count;
-                _version = 0;
```

Comment:
This is the constructor. Setting `_version` to 0 here is redundant because the CLR initializes all fields to their default values.